pytest-embedded/pytest_embedded/dut_factory.py

```diff
@@ -0,0 +1,655 @@
+    # otherwise, the process will still be here,
+    # but there are some problems with deletion, most likely because of the fork process.
+    # Call close instead delete.
+    def close(self):
+        while self.obj_stack:
+            layout = self.obj_stack.pop()
+            while layout:
+                obj = layout.pop()
+                _close_or_terminate(obj)
+            del layout
+        del self.obj_stack
+
+    def create_dut(self):
+        try:
+            layout = []
+            return self._create_dut(layout)
+        except Exception as e:
+            while layout:
+                obj = layout.pop()
+                _close_or_terminate(obj)
+            del layout
+            raise e
+
+    def _create_dut(self, layout):
+        global buff_parametrized_fixtures
+        msg_queue = msg_queue_gn()
+        layout.append(msg_queue)
+
+        _pexpect_logfile = os.path.join(
+            buff_parametrized_fixtures['_meta'].logdir, f'custom-dut-{dut_index_global}.txt'
+        )
+        logging.warning('You can get your custom DUT log file at the following path: %s.', _pexpect_logfile)
+
+        _listener = _listener_gn(msg_queue, _pexpect_logfile, True, dut_index_global, dut_index_global + 1)
+        layout.append(_listener)
+
+        _pexpect_fr = _pexpect_fr_gn(_pexpect_logfile, _listener)
+        layout.append(_pexpect_fr)
+
+        pexpect_proc = pexpect_proc_fn(_pexpect_fr)
+
+        kwargs = {
+            '_services': buff_parametrized_fixtures['_services'],
+            # parametrize fixtures
+            'app_path': '',
+            'build_dir': '',
+            'port': None,
+            'port_location': None,
+            'port_mac': None,
+            'target': None,
+            'beta_target': None,
+            'baud': None,
+            'skip_autoflash': None,
+            'erase_all': None,
+            'esptool_baud': None,
+            'part_tool': None,
+            'confirm_target_elf_sha256': None,
+            'erase_nvs': None,
+            'skip_check_coredump': None,
+            'panic_output_decode_script': None,
+            'openocd_prog_path': None,
+            'openocd_cli_args': None,
+            'gdb_prog_path': None,
+            'gdb_cli_args': None,
+            'no_gdb': None,
+            'qemu_image_path': None,
+            'qemu_prog_path': None,
+            'qemu_cli_args': None,
+            'qemu_extra_args': None,
+            'wokwi_cli_path': None,
+            'skip_regenerate_image': None,
+            'encrypt': None,
+            'keyfile': None,
+            # common
+            'test_case_name': buff_parametrized_fixtures['test_case_name'],
+            '_meta': buff_parametrized_fixtures['_meta'],
+            # pre-initialized fixtures
+            'dut_index': dut_index_global,
+            '_pexpect_logfile': _pexpect_logfile,
+            'pexpect_proc': pexpect_proc,
+            'msg_queue': msg_queue,
+        }
```

Comment:
Need to have some discuss about default values and how users can pass new values. I used the `buff_parametrized_fixtures` variable to store the previous fixture result.